src/separations.h

```diff
@@ -0,0 +1,130 @@
+#ifndef CYCAMORE_SRC_SEPARATIONS_H_
+#define CYCAMORE_SRC_SEPARATIONS_H_
+
+#include "cyclus.h"
+
+namespace cycamore {
+
+cyclus::Material::Ptr SepMaterial(std::map<int, double> effs, cyclus::Material::Ptr mat);
+
+class Separations : public cyclus::Facility {
+ public:
+  Separations(cyclus::Context* ctx);
+  virtual ~Separations(){};
+
+  virtual void Tick();
+  virtual void Tock();
+  virtual void EnterNotify();
+
+  virtual void AcceptMatlTrades(const std::vector<std::pair<
+      cyclus::Trade<cyclus::Material>, cyclus::Material::Ptr> >& responses);
+
+  virtual std::set<cyclus::RequestPortfolio<cyclus::Material>::Ptr>
+  GetMatlRequests();
+
+  virtual std::set<cyclus::BidPortfolio<cyclus::Material>::Ptr> GetMatlBids(
+      cyclus::CommodMap<cyclus::Material>::type& commod_requests);
+
+  virtual void GetMatlTrades(
+      const std::vector<cyclus::Trade<cyclus::Material> >& trades,
+      std::vector<std::pair<cyclus::Trade<cyclus::Material>,
+                            cyclus::Material::Ptr> >& responses);
+
+  #pragma cyclus clone
+  #pragma cyclus initfromcopy
+  #pragma cyclus infiletodb
+  #pragma cyclus initfromdb
+  #pragma cyclus schema
+  #pragma cyclus annotations
+  #pragma cyclus snapshot
+  // the following pragmas are ommitted and the functions are generated
+  // manually in order to handle the vector of resource buffers:
+  //
+  //     #pragma cyclus snapshotinv
+  //     #pragma cyclus initinv
+
+  virtual cyclus::Inventories SnapshotInv();
+  virtual void InitInv(cyclus::Inventories& inv);
+
+ private:
+  #pragma cyclus var { \
+    "doc" : "Maximum quantity of feed material that can be processed per time step.", \
+    "units": "kg", \
+  }
+  double throughput;
+
+  #pragma cyclus var { \
+    "doc": "Ordered list of commodities on which to request feed material to separate.", \
+    "uitype": ["oneormore", "incommodity"], \
+  }
+  std::vector<std::string> feed_commods;
+
+  #pragma cyclus var { \
+    "default": [], \
+    "doc": "Feed commodity request preferences for each of the given feed commodities (same order)." \
+           " If unspecified, default is to use zero for all preferences.", \
+  }
+  std::vector<double> feed_commod_prefs;
+
+  #pragma cyclus var { \
+    "doc": "Name for recipe to be used in feed requests." \
+           " Empty string results in use of an empty dummy recipe.", \
+    "uitype": "recipe", \
+    "default": "", \
+  }
+  std::string feed_recipe;
```

Comment:
This exists so that potential suppliers that provide based on the requested composition can be used with this facility.  For example, with no recipe at all, we just use a dummy recipe.  If a source facility that provides material according to requested compositions is the supplier (cycamore::Source is such a facility), then the source facility can do nothing except provide the dummy recipe, which is a very unuseful behavior.  So without this, separations would be constrained to only operate with facilities that provide what they provide regardless of what is requested.  This is the same reason that @mbmcgarry added a recipe option to the sink facility - so it can function with other facilities (e.g. enrichment) that require a meaningful recipe in a request.

I can't hardly imagine a separations facility that has more than 1 ideal input composition, and it otherwise doesn't care what it receives.  so a single, optional recipe is what I chose.